diff --git a/pysmt/cmd/install.py b/pysmt/cmd/install.py
index 016962c..68e2df5 100755
--- a/pysmt/cmd/install.py
+++ b/pysmt/cmd/install.py
@@ -32,10 +32,10 @@ from pysmt import __version__ as pysmt_version
 Installer = namedtuple("Installer",
                        ["InstallerClass", "version", "extra_params"])
 INSTALLERS = [
-    Installer(MSatInstaller,    "5.6.1", {}),
+    Installer(MSatInstaller,    "5.6.6", {}),
     Installer(CVC4Installer,    "1.7-prerelease",
               {"git_version" : "391ab9df6c3fd9a3771864900c1718534c1e4666"}),
-    Installer(Z3Installer,      "4.8.7", {"osx": "10.14.6"}),
+    Installer(Z3Installer,      "4.8.10", {"osx": "10.15.7"}),
     Installer(YicesInstaller,   "2.6.2",
               {"yicespy_version": "f0768ffeec15ea310f830d10878971c9998454ac"}),
     Installer(BtorInstaller,    "3.2.1", {}),
diff --git a/pysmt/cmd/installers/z3.py b/pysmt/cmd/installers/z3.py
index a72144a..6d6e8ce 100644
--- a/pysmt/cmd/installers/z3.py
+++ b/pysmt/cmd/installers/z3.py
@@ -31,7 +31,7 @@ class Z3Installer(SolverInstaller):
 
         system = self.os_name
         if system == "linux":
-            system = "ubuntu-16.04"
+            system = "ubuntu-18.04"
         elif system == "darwin":
             system = "osx-%s" % osx
         elif system == "windows":
